src/transformers/training_args.py

```diff
@@ -1522,6 +1540,16 @@ def __post_init__(self):
         if self.do_eval is False and self.eval_strategy != IntervalStrategy.NO:
             self.do_eval = True
 
+        if self.torch_empty_cache_steps is not None:
+            if not (isinstance(self.torch_empty_cache_steps, int) or self.torch_empty_cache_steps > 0):
+                raise ValueError(
+                    f"`torch_empty_cache_steps` must be an integer bigger than 0, got {self.torch_empty_cache_steps}."
+                )
+            if 0 < self.max_steps <= self.torch_empty_cache_steps:
+                raise ValueError(
+                    f"`torch_empty_cache_steps` must be smaller than `max_steps`, got torch_empty_cache_steps: {self.torch_empty_cache_steps}, max_steps: {self.max_steps}."
+                )
```

Comment:
I think we can remove this check. Although it makes sense: it's true that it won't have any effect if `torch_empty_cache_steps` > `max_steps`, but I can completely imagine someone iterating on configs quickly and just wanting to check `max_steps`.

Reply:
Would a warning be better then?

Reply:
I would just completely remove it

Reply:
alright, done